src/hooks/useWatch.ts

```diff
@@ -9,12 +9,13 @@ export function useWatch<
   },
   Name extends keyof T
 >(object: T, name: Name): T[Name] {
-  const [out, setOut] = React.useState<any>(object[name])
+  // We need to re-render the component when the value changes.
+  // Since [] !== [], we can use an empty array to force an update:
+  const [, rerender] = React.useState([])
```

Comment:
Wouldn't `NaN` work here because `NaN !== NaN`?

```
const [, rerender] = React.useState(NaN)
```

This feels more intentional of a hack. If you like the suggestion, then please try it and let me know if it works. Otherwise, what you have is fine.

Reply:
It won't work. React internally uses `Object.is` for comparisons, and `Object.is(NaN,NaN)` returns true.